azkaban-common/src/main/java/azkaban/executor/container/VPAFlowCriteria.java

```diff
@@ -42,33 +46,36 @@ public VPAFlowCriteria(final Props azkProps, final Logger logger) {
     this.fileLocation = azkProps.getString(Constants.
         ContainerizedDispatchManagerProperties.KUBERNETES_VPA_FLOW_FILTER_FILE, null);
     this.logger = logger;
-    loadFlowFilter(new HashMap<>());
+    loadFlowFilter();
   }
 
   /**
    * Reloads the in-memory flows map with the flow filter file.
    */
   public void reloadFlowFilter() {
     // Reset the filter map
-    loadFlowFilter(new HashMap<>());
+    loadFlowFilter();
   }
 
   @VisibleForTesting
   public void reloadFlowFilter(final String fileLocation) {
     // Reset the filter map
-    loadFlowFilter(new HashMap<>(), fileLocation);
+    loadFlowFilter(fileLocation);
   }
 
-  private void loadFlowFilter(final Map<String, Set<String>> flowMap) {
-    loadFlowFilter(flowMap, this.fileLocation);
+  private void loadFlowFilter() {
+    loadFlowFilter(this.fileLocation);
   }
 
   /**
    * Read the flow filter file and create a map of flow names. The read happens
    * on best effort basis. i.e, if a flow name is not correctly formatted, it
    * is ignored.
    */
-  private void loadFlowFilter(final Map<String, Set<String>> flowMap, final String fileLocation) {
+  synchronized private void loadFlowFilter(final String fileLocation) {
+    final Set<String> projectSet = new HashSet<>();
```

Comment:
It makes more sense to me to instantiate the objects closer to where they will be used.

Reply:
Yeah, that's why I moved it from outside of loadFlowFilter to inside.

Reply:
You create the objects, fileLocation is null, now you have a bunch of objects you're not going to use. Why not create before the try?

Reply:
Ah, I see your point now. Agree. Let me refactor this piece.

Reply:
done